lib/fizzy/execute.cpp

```diff
@@ -1376,13 +1378,13 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, span<const Value>
         }
         case Instr::i32_wrap_i64:
         {
-            stack.push(static_cast<uint32_t>(stack.pop()));
+            stack.push(stack.pop().as<uint32_t>());
             break;
         }
         case Instr::i64_extend_i32_s:
         {
-            const auto value = static_cast<int32_t>(stack.pop());
-            stack.push(static_cast<uint64_t>(int64_t{value}));
+            const auto value = stack.pop().as<int32_t>();
+            stack.push(int64_t{value});
```

Comment:
Do we actually need this `int64_t{}` here?

Reply:
Yes, it is needed as this is the core of this instruction implementation (otherwise for `int32_t` type the value will be cap to 32 bits). But it may be better to move it to the live above: `int64_t{stack.pop().as<int32_t>()}`.

Reply:
Could also squash it into a single one: `stack.push(int64_t{stack.pop().as<int32_t>()}`.

It all is a matter of preference, perhaps the current one is well balanced for readability.

Reply:
Fine to leave it too.